iis-wf/iis-wf-collapsers/src/main/java/eu/dnetlib/iis/wf/collapsers/basic/GenericCitationCollapser.java

```diff
@@ -30,18 +62,36 @@
 			list.add(citation);
 		}
 		List<Citation> results = new ArrayList<Citation>(citationsByPositionMap.size());
+		int citationsWithTextCount = 0;
 		for (List<Citation> citationsByPosition : citationsByPositionMap.values()) {
-			results.add(collapseForPosition(citationsByPosition));
+		    Citation collapsedCitation = collapseForPosition(citationsByPosition);
+		    if (hasTextDefined(collapsedCitation)) {
+		        citationsWithTextCount ++;
+		    }
+			results.add(collapsedCitation);
+		}
+		if (citationsWithTextCount > 0) {
+		    totalCounter.increment(citationsWithTextCount);
+		    docsWithAtLeastOneCitationCounter.increment(1);
 		}
 		return results;
     }
 
+    // --------------------- PRIVATE -------------------------------
+    
+    /**
+     * Checks whether text was defined for given citation.
+     */
+    private boolean hasTextDefined(Citation citation) {
+        return citation.getEntry().getRawText()!=null && StringUtils.isNotBlank(citation.getEntry().getRawText().toString());
```

Comment:
The orignal problem was `getRawText()` provides `CharSequence` while `org.apache.commons.lang.StringUtils.isNotBlank()` requires `String`.

Now I have realized `org.apache.commons.lang3.StringUtils.isNotBlank()` is superior to `org.apache.commons.lang.StringUtils.isNotBlank()` because it accepts `CharSequence` what perfectly suits our needs :)